### PR TITLE
Make narrator read variable explorer and package manager

### PR DESCRIPTION
**Pull request by MikhailArkhipov**

### CODE REVIEW

File: src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml.cs [hunk 1]
```diff
@@ -88,5 +91,17 @@ public partial class PackageList : UserControl {
             var package = GetPackage(e);
             Model?.UnloadAsync(package).DoNotWait();
         }
+
+        protected override AutomationPeer OnCreateAutomationPeer() {
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml.cs]: Why do you need it? `UserControl` has an `AutomationPeer` already

File: src/Package/Impl/DataInspect/VariableView.xaml.cs [hunk 2]
```diff
@@ -215,6 +215,11 @@ public partial class VariableView : UserControl, ICommandTarget, IDisposable {
                 DeleteCurrentVariableAsync().DoNotWait();
             } else if (e.Key == Key.Apps) {
                 ShowContextMenu();
+            } else if(e.Key == Key.Space) {
```
Comment (AlexanderSher) [inline src/Package/Impl/DataInspect/VariableView.xaml.cs]: Do we support any kind of editing in `VariableView`?

File: src/Package/Impl/DataInspect/VariableView.xaml [hunk 3]
```diff
@@ -83,7 +83,8 @@
                     <ToggleButton x:Name="Expander"
                               Grid.Column="1"
                               Style="{StaticResource ExpanderToggleStyle}"
-                              ClickMode="Press" IsChecked="{Binding IsExpanded, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" />
+                              ClickMode="Press" IsChecked="{Binding IsExpanded, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" 
+                              AutomationProperties.Name="Expand variable"/>
```
Comment (AlexanderSher) [inline src/Package/Impl/DataInspect/VariableView.xaml]: This should be a resource

File: src/Package/Impl/DataInspect/REnvironment.cs [hunk 4]
```diff
@@ -40,5 +40,8 @@ public REnvironment(string name)
         public string EnvironmentExpression { get; }
 
         public REnvironmentKind Kind { get; }
+
+        // For Accessibility: screen reader reads values returned by 'ToString()'
+        public override string ToString() => Name ?? base.ToString();
```
Comment (AlexanderSher) [inline src/Package/Impl/DataInspect/REnvironment.cs]: I'd rather add `AutomationProperties.Name` to the `EnvironmentComboBox.ItemContainerStyle` style

File: src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml.cs]: To fire property change event so narrator re-reads the state indicating that the package is now loaded or installed.

File: src/Package/Impl/DataInspect/VariableView.xaml.cs (hunk 2 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/DataInspect/VariableView.xaml.cs]: Nope

File: src/Package/Impl/DataInspect/VariableView.xaml (hunk 3 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/DataInspect/VariableView.xaml]: Good catch

File: src/Package/Impl/DataInspect/REnvironment.cs (hunk 4 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/DataInspect/REnvironment.cs]: For some reason that didn't work for me, but let me try again then

File: src/Package/Impl/DataInspect/VariableView.xaml.cs (hunk 2 above)
Comment (AlexanderSher) [inline src/Package/Impl/DataInspect/VariableView.xaml.cs]: Ok

**Comment by MikhailArkhipov** [comment]
Will bind `AutomationProp.Name` in XAML for package manager as well.

File: src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml.cs (hunk 1 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml.cs]: But we already fire `PropertyChanged` when `IsInstalled` or 'IsLoaded` are changing. Does automation skip them for some reason?

File: src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml [hunk 5]
```diff
@@ -153,6 +153,7 @@
             <!-- the style of the ListBoxItem when the item is a package. -->
             <Style x:Key="PackageItemStyle" TargetType="{x:Type ListBoxItem}" BasedOn="{StaticResource FillingListBoxItemStyle}">
                 <EventSetter Event="MouseDoubleClick" Handler="List_MouseDoubleClick" />
+                <Setter Property="AutomationProperties.Name" Value="{Binding AccessibleDescription}" />
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml]: You can split `AccessibleDescription` between `AutomationProperties.Name` and `AutomationProperties.HelpText`. This will allow user to skip the details if he doesn't want to hear them.

Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml]: Up to you.

File: src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/View/PackageList.xaml.cs]: Actually 'FocusChange`. It doesnt re-read on property change.

**Comment by AlexanderSher** [comment]
👍
